sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1415,15 +1425,22 @@ bool gnrc_netif_ipv6_wait_for_global_address(gnrc_netif_t *netif,
 
     /* wait for global address */
     msg_t m;
-    ztimer_now_t t_stop = ztimer_now(ZTIMER_MSEC) + timeout_ms;
+    ztimer_t _timeout = { .callback = _unset_arg, .arg = &_timeout };
+    if (timeout_ms != UINT32_MAX) {
+        ztimer_set(ZTIMER_MSEC, &_timeout, timeout_ms);
+    }
+
     while (!has_global) {
         /* stop if timeout reached */
-        if (ztimer_now(ZTIMER_MSEC) > t_stop) {
+        if (!_timeout.arg) {
             DEBUG_PUTS("gnrc_netif: timeout reached");
             break;
         }
+
         /* waiting for any message */
-        if (ztimer_msg_receive_timeout(ZTIMER_MSEC, &m, timeout_ms) < 0) {
+        if (timeout_ms == UINT32_MAX) {
+            msg_receive(&m);
+        } else if (ztimer_msg_receive_timeout(ZTIMER_MSEC, &m, timeout_ms) < 0) {
```

Comment:
```suggestion
        }
        else if (ztimer_msg_receive_timeout(ZTIMER_MSEC, &m, timeout_ms) < 0) {
```
according to RIOT coding style

Reply:
since both styles are already in this file, this can be ignored